codegen/template.go

```diff
@@ -81,9 +99,17 @@ type Template struct {
 
 // NewTemplate creates a bundle of templates.
 func NewTemplate() (*Template, error) {
```

Comment:
Nit: I would rename
`NewTemplate`-> `NewDefaultTemplate`
`NewTemplateFromAssetProvider` -> `NewTemplate`, because the arguments type says `FromAssetProvider`